CesiumUtility/include/CesiumUtility/JsonValue.h

```diff
@@ -268,8 +275,53 @@ class CESIUMGLTF_API JsonValue final {
     return it->second.getSafeNumber<To>();
   }
 
+  /**
+   * @brief Converts the numerical value corresponding to the given key
+   * to the provided numerical template type.
+   *
+   * If this instance is not a {@link JsonValue::Object}, the key does not exist
+   * in this object, or the named value does not have a numerical type that can
+   * be represented as T without precision loss, then the default value is
+   * returned.
+   *
+   * @tparam To The expected type of the value.
+   * @param key The key for which to retrieve the value from this object.
+   * @return The converted value.
+   * @throws If unable to convert the converted value for one of the
+   * aforementioned reasons.
+   * @remarks Compilation will fail if type 'To' is not an integral / float /
+   * double type.
+   */
+  template <
+      typename To,
+      typename std::enable_if<
+          std::is_integral<To>::value ||
+          std::is_floating_point<To>::value>::type* = nullptr>
+  [[nodiscard]] To getSafeNumericalValueOrDefaultForKey(
```

Comment:
It's not necessarily a problem to have methods that throw. They're perhaps preferred when exceptions are enabled, and they're valid even when exceptions disabled if you happen to _know_ the method won't be throwing in your particular case. A throw statement will still compile with exceptions disabled, it will just cause immediate program termination if it's executed.

The bigger problem we had was the use of a try/catch block. Under some compilers, at least (VS2017), that code won't even compile when exceptions are disabled. And even if it did, it presumably would not work as expected because the non-throwing contract of the method wouldn't have been able to work without the ability to throw and catch exceptions.

I think we may want to do another round of refactoring JsonValue in the future, but for now it's ok to have these throwing methods given that we also have non-throwing alternatives.